src/components/App.jsx

```diff
@@ -1,44 +1,78 @@
 /* global cozy */
-import React, { useState, useEffect } from 'react'
+import React, { useState, useEffect, useCallback } from 'react'
 import { PropTypes } from 'prop-types'
 import flow from 'lodash/flow'
 
+import { useClient } from 'cozy-client'
 import { Main, Layout } from 'cozy-ui/transpiled/react/Layout'
 import Alerter from 'cozy-ui/transpiled/react/Alerter'
 import { Title } from 'cozy-ui/transpiled/react/Text'
 import { useI18n, useBreakpoints } from 'cozy-ui/transpiled/react'
 import { Sprite as IconSprite } from 'cozy-ui/transpiled/react/Icon'
 import flag, { FlagSwitcher } from 'cozy-flags'
 import 'cozy-ui/transpiled/react/stylesheet.css'
+import log from 'cozy-logger'
 
 import withContactsMutations from '../connections/allContacts'
 import ContactsSelectionBar from './layout/ContactsSelectionBar'
 import { ModalManager } from '../helpers/modalManager'
 import { initFlags } from '../helpers/flags'
 import container from './AppContainer'
 import ContentWrapper from './ContentWrapper'
+import SpinnerContact from './Components/Spinner'
+import { fetchNormalizedServiceByName } from '../helpers/fetches'
 
 const ContactsApp = props => {
   // HACK to avoid CozyBar error :
   // you tried to use the CozyBar API (BarCenter) but the CozyBar is not initialised yet via cozy.bar.init
   // TODO : TO BE REMOVED
   const [cozyBarHack, setcozyBarHack] = useState(false)
 
-  useEffect(() => {
-    initFlags()
-    props.cleanTrashedGroups()
+  const [serviceToLaunch, setServiceToLaunch] = useState(null)
+  const [hasServiceBeenLaunched, setHasServiceBeenLaunched] = useState(null)
 
-    // HACK to be removed
-    setTimeout(() => {
-      setcozyBarHack(true)
-    }, 0)
+  const client = useClient()
+
+  const setStateOfServiceToLaunch = useCallback(async () => {
+    const serviceToLaunch = await fetchNormalizedServiceByName(
+      client,
+      'keepIndexFullNameAndDisplayNameUpToDate'
+    )
+    setServiceToLaunch(serviceToLaunch)
+    setHasServiceBeenLaunched(
+      serviceToLaunch.current_state.last_success.length > 0
+    )
   }, [])
 
+  useEffect(
+    () => {
+      initFlags()
+      props.cleanTrashedGroups()
+      setStateOfServiceToLaunch()
+
+      // HACK to be removed
+      setTimeout(() => {
+        setcozyBarHack(true)
+      }, 0)
+    },
+    [setStateOfServiceToLaunch]
```

Comment:
To me, `useEffect` should serve only one purpose, not several. Specially when several methods in this useEffect do not share the dependency. It the identity of `setStateOfServiceToLaunch` changes, the effect will fire again. 

By creating a custom hook for each purpose, it's easier to unit test them, and prevent unintended executions.